packages/network/lib/uri.ts

```diff
@@ -87,3 +87,15 @@ export function addDefaultPort (urlToCheck) {
 export function getPath (urlToCheck) {
   return url.parse(urlToCheck).path
 }
+
+const localhostIPRegex = /^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/
+
+export function isLocalhost (url: URL) {
```

Comment:
what about `localhost6`? :stuck_out_tongue: i'm mostly kidding... but i do think that this area might be error-prone, especially since i don't see any links to RFCs or browser source code showing how they do this heuristic. do you have any references for how browsers decide this?

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/20450/commits/a2f7bd8b10935299f0bd05ddc648cc9e70d97a4c

Reply:
There is similar code in https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/location.ts#L16 for verifying localhost as well. Seems like it would be pretty simple to update that code to use this.

Reply:
that regex you linked actually looks wrong, i don't think `0.0.0.0` resolves to localhost

Reply:
I don't know why it's specifically checking for localhost there. Seems like it should just be a general IP address check since it's trying to determine if it's a url it can normalize.

Reply:
I believe it's addressing this requirement:

```
Cypress automatically prepends the http:// protocol to common hosts. If you're not using one of these 3 hosts, then make sure to provide the protocol.

cy.visit('localhost:3000') // Visits http://localhost:3000
cy.visit('0.0.0.0:3000') // Visits http://0.0.0.0:3000
cy.visit('127.0.0.1:3000') // Visits http://127.0.0.1:3000
```

https://docs.cypress.io/api/commands/visit#Protocol

Reply:
Ah good find. In that case, it seems like that variable name is a bit misleading, but we probably shouldn't change the behavior since it's documented that way.

Reply:
Except for maybe `0.0.0.0`... I don't think there's any possible way to load something from the null route